crates/node-api/src/evm.rs

```diff
@@ -0,0 +1,73 @@
+use reth_primitives::{
+    revm::env::{fill_block_env_with_coinbase, recover_header_signer},
+    Address, Chain, ChainSpec, Header, Transaction, TransactionSignedEcRecovered, U256,
+};
+use revm_primitives::{BlockEnv, CfgEnv, SpecId, TxEnv};
+
+/// This represents the set of methods used to configure the EVM before execution.
+pub trait EvmEnvConfig {
+    /// Additional metadata that can be used when filling the transaction environment.
+    type TxMeta;
```

Comment:
I think this is fine now, though if we move away from storing the encoded bytes in the txenv this will be redundant